olp-cpp-sdk-core/src/client/repository/ApiCacheRepository.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "ApiCacheRepository.h"
+
+#include <olp/core/cache/KeyValueCache.h>
+#include <olp/core/logging/Log.h>
+
+namespace {
+constexpr auto kLogTag = "ApiCacheRepository";
+constexpr time_t kLookupApiExpiryTime = 3600;
```

Comment:
Let's take it from the cache-control header first and then default value only if cache-control is not set.

Reply:
done
ApiCacheRepository now has expiry input for `Put()`
`PlatformApi` and `ResourceApi` returns expiry(if found) with apis.